Rename redis-port to port and add default

mfitton opened the pull request:
<!-- Thank you for your contribution! Please review https://github.com/ray-project/ray/blob/master/CONTRIBUTING.rst before opening a pull request. -->

## Why are these changes needed?
The port we expose the Ray service for connections on being the port that we start Redis on is an implementation detail that users shouldn't have to know.
Here we add a default port option so that users will have consistency across runs even if they don't set an argument, as well as a fallback to a random port like we used to use. This makes it possible to omit the port argument when setting up a multi-node cluster.

## Related issue number
https://anyscale.atlassian.net/browse/RS-1370

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [x] This PR is not tested with code (please justify below)

I tested this PR by running the command with the relevant new configurations and combinations of them. There are no existing tests for the scripts submodule.

AmplabJenkins commented:
Can one of the admins verify this patch?

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Ray-PRB/25789/
Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Ray-PRB/25795/
Test PASSed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Ray-PRB/25892/
Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Ray-PRB/25898/
Test FAILed.

edoakes commented on python/ray/ray_constants.py:
```diff
@@ -32,6 +32,9 @@ def env_bool(key, default):
 REDIS_MINIMUM_MEMORY_BYTES = 10**7
+# If a user does not specify a port for the primary Ray service,
+# we attempt to start the service running at this port.
+DEFAULT_PORT = 52600
```
We should match this to the current defaults we use in the autoscaler configs, 6379:
https://github.com/ray-project/ray/blob/master/python/ray/autoscaler/aws/example-full.yaml#L137

mehrdadn commented:
@mfitton Regarding linting: would you mind merging in master and seeing if you still get the lint errors? It seems like you haven't touched those files at all, so they shouldn't get linted if they don't show up in the diff.

If you do that and still get the error with an up-to-date master, then if the GitHub Actions linter is in _conflict_ with the Travis one, please comment out the `. ./ci/travis/ci.sh lint` line in `.github/workflows/main.yml`. Otherwise, if it's not actually in conflict, would you mind just making the changes it suggests?

In all cases, let's make sure the builds are passing before merging. Thanks!

mfitton commented:
Thanks everyone for the feedback.
@edoakes I changed the cluster yaml files and changed the default port.
@simon-mo I updated those docs
@mehrdadn I merged in master, I'll let you know how the linting goes from there.

mfitton commented:
@mehrdadn the build failed in the GitHub actions linter, which still says it is using version ~9.1 of clang-format instead of version 7.0.0 which the log says Ray is supposed to use.
The travis build is still queued.

mehrdadn commented:
@mfitton Oh nice, the previous errors went away. Now it's just a couple minor issues in the same files you're editing. If you edit it and make the [changes shown](https://github.com/ray-project/ray/pull/8406/checks?check_run_id=679084156#step:5:440) you should be good to go!

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Ray-PRB/25935/
Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Ray-PRB/25937/
Test FAILed.

edoakes commented:
@mfitton tests look good except lint

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Ray-PRB/26041/
Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Ray-PRB/26043/
Test PASSed.
